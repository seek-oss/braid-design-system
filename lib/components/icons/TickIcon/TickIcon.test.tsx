import React from 'react';
import { TickIcon } from '../..';
import { render } from 'react-testing-library';
import forEachTheme from '../../../test/utils/forEachTheme';

describe('TickIcon', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('Default', () => {
      const { container } = render(
        <ThemeProvider>
          <TickIcon />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Sized', () => {
      const { container } = render(
        <ThemeProvider>
          <TickIcon size="large" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Inline', () => {
      const { container } = render(
        <ThemeProvider>
          <TickIcon inline />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
