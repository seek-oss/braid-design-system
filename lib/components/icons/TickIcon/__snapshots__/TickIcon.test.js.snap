// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickIcon Theme: jobStreet Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: jobStreet Inline 1`] = `
<svg
  class="Icon__inline width__standardTextInline height__standardTextInline Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: jobStreet Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: seekAnz Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: seekAnz Inline 1`] = `
<svg
  class="Icon__inline width__standardTextInline height__standardTextInline Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: seekAnz Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: seekAsia Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: seekAsia Inline 1`] = `
<svg
  class="Icon__inline width__standardTextInline height__standardTextInline Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: seekAsia Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: wireframe Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: wireframe Inline 1`] = `
<svg
  class="Icon__inline width__standardTextInline height__standardTextInline Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;

exports[`TickIcon Theme: wireframe Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__default"
  height="16"
  viewBox="90.471 302.809 823 823"
  width="16"
>
  <path
    d="M365.407 1022.185L90.677 747.457l67.883-67.882L365.406 886.42l479.977-479.988 67.883 67.882"
  />
</svg>
`;
